### ISSUE TITLE
Multiple "Referenced element is not defined" errors in many platform bundles

### ISSUE THREAD

**Issue by iloveeclipse** [eclipse-pde/eclipse.pde#890]
Looks like a severe regression from aa208c02423ac8aa48f10a14d5c2003b2147d1ce / https://github.com/eclipse-pde/eclipse.pde/pull/845.

Last good build seem to be `I20231030-0450`

I see now many Plug-In errors like `Referenced element 'and' is not defined` reported for:
- org.eclipse.debug.ui
- org.eclipse.ui.console
- org.eclipse.tips.core
- org.eclipse.ui.intro
- org.eclipse.jdt.core.manipulation
- org.eclipse.jdt.ui
- org.eclipse.ui
- org.eclipse.ui.ide
- org.eclipse.ui.navigator
- org.eclipse.core.resources
- org.eclipse.core.expressions
- org.eclipse.pde.ui.tests
- org.eclipse.core.expressions.tests
- may be more, have not all bundles in the workspace

![image](https://github.com/eclipse-pde/eclipse.pde/assets/964108/8af36ede-b661-4da6-94ba-997d4b641fe3)

**Comment by laeubi** [comment]
I don't see this error in

```
Version: 2023-12 (4.30)
Build id: I20231102-1800
```

**Comment by iloveeclipse** [comment]
Looks like you need a clean build on affected projects. That seem to be the reason I haven't noticed the errors before - I always use last nightly build for my day work.

I can reproduce with Eclipse started on Java 21 and on Java 17, both in the old and in a fresh new workspace:

![image](https://github.com/eclipse-pde/eclipse.pde/assets/964108/96dcb18c-7d4f-4937-a65a-e92fdf338f07)

**Comment by merks** [comment]
I expect it's being caused by the include not being processed properly:
```
<include schemaLocation="schema://org.eclipse.core.expressions/schema/expressionLanguage.exsd"/>
```

![image](https://github.com/eclipse-pde/eclipse.pde/assets/208716/e8c12b30-4cd7-4f24-aa62-3a621cfe3efb)

**Comment by laeubi** [comment]
> Looks like you need a clean build on affected projects

This did the trick for me, will look into it.

**Comment by merks** [comment]
I Oomphed a fresh installation with just the Platform UI clone and here a clean build does not fix the problem:

![image](https://github.com/eclipse-pde/eclipse.pde/assets/208716/ab165b5d-9773-48b0-a24c-74575989fdde)

**Comment by laeubi** [comment]
> Sorry, I misread. No one claimed a clean build fixes the problem. Sorry.

To make it clear, you need a clean build to **show the problem** because there seems to be some sort of aching involved.

**Comment by laeubi** [comment]
I think I found the cause for this, but I'm a bit confused if there is maybe an existing bug, can anyone confirm that double click the included schema has worked before?

Even if the included schema is found (no more red `x`) the double click fails to open the editor, this is because the location starts with schema:// (what seems intentional) but then the method  `SchemaEditor.openSchema( ..)`  fails because it assumes the first part of the path is the plugin id what obviously is wrong here when the path is `schema://org.eclipse.core.expressions/schema/expressionLanguage.exsd` it is always (at least on linux) `schema`

![grafik](https://github.com/eclipse-pde/eclipse.pde/assets/1331477/853c581a-f48b-4c1e-a12f-feb1cc1806ca)
![grafik](https://github.com/eclipse-pde/eclipse.pde/assets/1331477/fc34c360-293c-4a55-8be2-70765c6b1d96)

I think it can work on windows because maybe there the `:` is interpreted as not being part of the path ...

**Comment by iloveeclipse** [comment]
> can anyone confirm that double click the included schema has worked before?

Yes, I can see it was not working already in the build where we don't have schema errors:

![image](https://github.com/eclipse-pde/eclipse.pde/assets/964108/f232dc98-7ac5-4502-b55f-f71287ba92bb)

**Comment by merks** [comment]
On Windows the schema: is the device, not the 0th segment:

![image](https://github.com/eclipse-pde/eclipse.pde/assets/208716/abcf9218-08d2-4da0-9b31-045b1fbaa9db)

**Comment by laeubi** [comment]
Thanks for confirmation I now created
- https://github.com/eclipse-pde/eclipse.pde/pull/891

that should fix the issue under Linux with open an editor as well as with missing included schemas.

**Comment by jpstotz** [comment]
> I expect it's being caused by the include not being processed properly:
> 
> ```
> <include schemaLocation="schema://org.eclipse.core.expressions/schema/expressionLanguage.exsd"/>
> ```
> 
> ![image](https://github.com/eclipse-pde/eclipse.pde/assets/208716/e8c12b30-4cd7-4f24-aa62-3a621cfe3efb)

I now have the exact same problem using Eclipse 2025-06 on Windows 11.
